CONTRIBUTING.md

```diff
@@ -113,6 +113,34 @@ exist to show coverage directly in your editor. E.g.:
 - <https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters>
 - <https://github.com/alepez/vim-llvmcov>
 
+### Documentation
+
+Here are some guidelines for writing documentation in Ratatui.  
+Every public API should be documented.
+
+#### Content
+
+The main doc comment should talk about the general features that the widget supports and introduce the concepts pointing to the various methods. Focus on interaction with various features and giving enough information that helps understand why you might want something.  
```

Comment:
Not sure what you mean. Required line length is specified in the *Format* section. Should I move it over here?

Reply:
Sorry for being too brief. We have a markdownlint rule for 100 characters to make it easier to edit, merge, diff etc. The smiley was because this was humorous appearing in a guidelines change.